src/Phobos.Kafka/src/Petabridge.Phobos.Kafka.Producer/AkkaService.cs

```diff
@@ -113,6 +153,60 @@ public AkkaActors(ActorSystem sys)
         public IActorRef RouterForwarderActor { get; }
         
         public IActorRef ProducerActor { get; }
+
+        private Source<IResults<TKey, TValue, NotUsed>, IActorRef> ProducerActorRefWithCommit<TKey, TValue>(
+            ActorSystem system, 
+            string topic, 
+            ProducerSettings<TKey, TValue> producerSettings)
+        {
+            return Source.ActorRef<(ISpanContext spanContext, TKey key, TValue message)>(128, OverflowStrategy.DropNew)
+                .Select(tuple =>
+                {
+                    var (spanContext, key, message) = tuple;
+                    var record = new ProducerRecord<TKey, TValue>(topic, key, message).WithTracing(system, spanContext);
+                    return ProducerMessage.Single(record);
+                })
+                .Via(KafkaProducer.FlexiFlow<TKey, TValue, NotUsed>(producerSettings));
+        }
+        
+        private Source<IResults<TKey, TValue, NotUsed>, IActorRef> ProducerActorRefWithCommit<TKey, TValue>(
```

Comment:
Not used in this sample, an example code of a generic way for creating an actor ref source with commitable kafka mesage, passing in an actual kafka record instead of the message value.